src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -728,18 +729,17 @@ void InterpreterCreateQuery::setEngine(ASTCreateQuery & create) const
         if (create.temporary && create.storage && create.storage->engine && create.storage->engine->name != "Memory")
             throw Exception(ErrorCodes::INCORRECT_QUERY,
                 "Temporary tables can only be created with ENGINE = Memory, not {}", create.storage->engine->name);
-
-        return;
```

Comment:
Seems like without this `return` statement we will overwrite explicitly specified engine in `CREATE AS` query.

Reply:
I've checked it locally and explicitly specified engine is not overwritten. However, it's not clear from code and it's not clear why this `return` statement is removed. Consider using [return early pattern](https://szymonkrajewski.pl/why-should-you-return-early/) pattern here and removing all subsequent `if (!create.storage)` conditions.

Reply:
Tried to simplify logic there. BTW previous one was also not perfect